tools/cpplint.py

```diff
@@ -709,8 +709,9 @@ def RepositoryName(self):
     locations won't see bogus errors.
     """
     fullname = self.FullName()
-    # XXX(bnoordhuis) Expects that cpplint.py lives in the tools/ directory.
-    toplevel = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+    toplevel = os.path.abspath(
+        os.path.join(os.path.dirname(__file__), '..')
+    ).replace('\\', '/')
```

Comment:
@bnoordhuis Maybe I am not understanding the comment. As far as I can tell `cpplint.py` is located in the tools directory. The `toplevel` var expects the current file to live in the `tools/` directory. According to #4635 I removed the comment if everything checks out. Unless, by toplevel you are expecting the name of the root directory i.e. `node/` 
or the repo dir.

```python
def FullName(self):
    """Make Windows paths like Unix."""
    return os.path.abspath(self._filename).replace('\\', '/')

  def RepositoryName(self):
    """FullName after removing the local path to the repository.
    If we have a real absolute path name here we can try to do something smart:
    detecting the root of the checkout and truncating /path/to/checkout from
    the name so that we get header guards that don't include things like
    "C:\Documents and Settings\..." or "/home/username/..." in them and thus
    people on different computers who have checked the source out to different
    locations won't see bogus errors.
    """
    fullname = self.FullName()
    # XXX(bnoordhuis) Expects that cpplint.py lives in the tools/ directory.
    toplevel = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
    prefix = os.path.commonprefix([fullname, toplevel])
    return fullname[len(prefix) + 1:]
```
At the end there with the `commonprefix` along with the returned splice, I interpreted that as you wanting to delete everything before the tools directory (this includes the repo name/dir) which is why I took toplevel to actually mean the tools dir and not the repo dir. However, now it doesn't make sense with the name of the method being `RepositoryName`.  Can I get some clarification here?


@thefourtheye `os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))` yields a path to node or the repo directory where as `os.path.abspath(os.path.dirname(__file__))` gives the path to the `tools/` directory. It serves as an alias for `cd ..` but for file strings. Am I missing something here?

With that being said I guess we would **actually** return the repo name as suggested by the method name with `return prefix.split('/')[-1]` which would return `node` the repo name instead of `fullname[len(prefix) + 1]` which returns `tools/cpplint.py`.  And judging from your doc comment it looks like `return os.path.join(prefix.split('/')[-1], fullname[len(prefix) + 1:]).replace('\\', '/')` will give `node/tools/cpplint.py` which is consistent with...

> Fullname after removing the local path to the repository